app/src/main/java/com/example/tasky/authentication/presentation/signup/viewmodel/SignUpViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.tasky.authentication.presentation.signup.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class SignUpViewModel @Inject constructor() : ViewModel() {
+
+    var state by mutableStateOf(SignUpState())
+        private set
+
+    fun onEvent(event: SignUpEvent) {
+        when (event) {
+            is SignUpEvent.OnEmailValueChanged -> {
+                state = state.copy(emailValue = event.emailValue)
+            }
+
+            is SignUpEvent.OnNameValueChanged -> {
+                state = state.copy(nameValue = event.nameValue)
+            }
+
+            is SignUpEvent.OnPasswordValueChanged -> {
+                state = state.copy(passwordValue = event.passwordValue)
+            }
+
+            is SignUpEvent.OnShowPasswordValueChanged -> {
+                state = state.copy(showPassword = event.showPassword)
+            }
+
+            SignUpEvent.OnSignUpClicked -> {
+                // TODO
+                state = state.copy(onSignUpSucceed = true)
+            }
+
+            SignUpEvent.BackNavigated -> state = state.copy(navigateBack = null)
+            SignUpEvent.ErrorShown -> state = state.copy(errorMessage = null)
+            SignUpEvent.OnBackClicked -> state = state.copy(navigateBack = null)
+        }
+    }
+}
+
+data class SignUpState(
```

Comment:
Personally I like to have the State and Event as their own classes
I find them easier to find